CIF-383 - Add issue and PR templates to public repos

masodu opened the pull request:
I plan to add issue and PR templates to all our repos. Please review this and once we agree on a final version I will add it to the other repos as well.

herzog31 commented on ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,13 @@
+### Expected Behaviour
```
I think this issue template works for bugs, but not for actual feature requests. Maybe we should restructure the template a bit to have one for bugs and one for stories.

masodu replied:
Depends on how you look at this. An expected behaviour can also be something new. Any proposal for the story template?

herzog31 replied:
What about the following:

### User Story
<!--- Add a user story in the format -->
<!--- As a [user], I want [goal] so that [reason]. -->

### Description & Motivation
<!--- Describe the feature and name the business value of it. -->

### Deliverables
<!--- What needs to be done? Example: -->
<!--- Code changes -->
<!--- Documentation update -->
<!--- Follow up issues -->

### Acceptance Criteria
<!--- A list of criteria to be used as requirements for implementation. Example: -->
<!--- Multiple languages are supported -->
<!--- The description field accepts and verifies HTML -->
<!--- Maximum allowed time for responding is 200ms. -->

### Verification Steps
<!--- Steps used to verify this story, including instruction for setup. -->

masodu commented:
@herzog31 IMO we should leave all these points as they serve as checklist. The PR will not going to pass, I agree, but at the same time is good to make aware the author what should be done before creating PRs. This way we can avoid having many failed PRs. WDYT?

herzog31 commented:
@smaftei Okay, that makes sense. I would make the checklist more concrete then. For example:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code passes the code style as defined by ESLint.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all integration tests locally and they pass.

masodu commented:
- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code passes the code style as defined by ESLint.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

I don't think that the user can actually

herzog31 commented on FEATURE_TEMPLATE.md:
```diff
@@ -0,0 +1,21 @@
+### User Story
```
btw I just read https://help.github.com/articles/creating-issue-templates-for-your-repository/
I think we should configure it like this and not add these files, wdyt?

herzog31 replied:
Yes, but if you configure it that way, you will be asked which template to use when you create an issue. Feel free to try it. I haven't seen it in action yet, just found the GitHub documentation.

masodu replied:
You need to have the correct file names - this is why I opened it - to force user go through a checklist :). I am sure about ISSUE and PR but not sure about Feature. We need to test that .

masodu commented:
@herzog31 I removed the feature template as with git hub we can open only issues and prs. Maybe it was not clear but these will be automatically pulled by git when issues / pr will be opened. I also moved all these files to .github. WDYT?

dplaton commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,13 @@
+
+### Actual Behaviour
+
+### Reproduce Scenario (including but not limited to)
```
What's the different between "Reproduce scenario" and "Steps to reproduce" below?

masodu replied:
I think one is summary one are the exact details. Templates were provided by open source guys.

dplaton commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -0,0 +1,46 @@
+<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
+<!--- Please link to the issue here: -->
+
+## Motivation and Context
```
This could (should?) be part of the issue.

masodu replied:
I would say that the issue context and motivation is clear. I guess opening a PR should basically be linked to some issue / feature .

herzog31 commented:
@masodu please read again the documentation I linked, I think you didn't understand what it was describing. I set it up in a repo of mine as example, please check https://github.com/herzog31/locust/tree/master/.github/ISSUE_TEMPLATE.

masodu commented:
@herzog31 please check again.
